src/MongoDB/Session.c

```diff
@@ -246,6 +234,31 @@ static PHP_METHOD(Session, getOperationTime)
 	php_phongo_new_timestamp_from_increment_and_timestamp(return_value, increment, timestamp TSRMLS_CC);
 } /* }}} */
 
+/* {{{ proto MongoDB\Driver\Server|null MongoDB\Driver\Session::getServer()
+   Returns the server this session is pinned to */
+static PHP_METHOD(Session, getServer)
+{
+	php_phongo_session_t* intern;
+	uint32_t              server_id = 0;
+
+	intern = Z_SESSION_OBJ_P(getThis());
+	SESSION_CHECK_LIVELINESS(intern, "getOperationTime")
+
+	if (zend_parse_parameters_none() == FAILURE) {
+		return;
+	}
+
+	server_id = mongoc_client_session_get_server_id(intern->client_session);
+
+	/* For sessions without a pinned server, 0 is returned. In this case, we
+	 * return null to force a server selection round. */
+	if (server_id == 0) {
```

Comment:
FWIW, `php_phongo_manager_select_server()` does `if (selected_server)`, so we could just do `(!server_id)` here if you prefer. It's not as sloppy to do that in C as it would otherwise be in PHP itself :)